### PR TITLE
Feature/database structure

### PR DESCRIPTION
**Pull request by gchilczuk**
Ready for review

### CODE REVIEW

File: game/models.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+AnsResp = namedtuple('AnsResp', ['correct', 'feedback'])
+
+
+class Reward(models.Model):
+    """Representation of the reward of the game"""
+    description = models.TextField()
+
+
+class Edition(models.Model):
```
Comment (marwaw) [inline game/models.py:16]: I don't get the name :|

File: game/models.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    finish = models.DateTimeField(null=True)
+    reward = models.ForeignKey(Reward, on_delete=models.SET_NULL, null=True)
+
+
+class Quest(models.Model):
+    """
+    Represents basic quest entity.
+    Player must complete the task to get the next one.
+    To confirm that quest is solved player have to
+    correctly answer the final question of the question.
```
Comment (marwaw) [inline game/models.py]: `of the quest`

File: game/models.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    :id: uuid, primary key; universally unique identifier makes quests ordered non-sequentialy
+    :title: the title of the quest
+    :description: detailed description of quest
+    :question: final question of the quest
+    :answer:
+    """
```
Comment (marwaw) [inline game/models.py]: Missing fields description

File: game/models.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    title = models.CharField(max_length=32)
+    description = models.TextField(blank=True)
+    question = models.CharField(max_length=100)
+    answer = models.OneToOneField('Answer', models.SET_NULL, null=True)
+    succeeding = models.ForeignKey('Quest', on_delete=models.SET_NULL, related_name='previous', null=True)
+    edition = models.ForeignKey(Edition, models.CASCADE, related_name='quests', related_query_name='quest')
```
Comment (marwaw) [inline game/models.py]: It's easier to read if you add key-word arguments

File: game/models.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    :answer: correct answer of final question of the Quest
+    :confirmation: feedback provided when user provide correct answer
+    :default: default rejection response
+    :rejections: dictionary of special-case responses to wrong answers.
+        It may be useful for example when is very probable that user will misspell answer,
+        or when answer is close enough to provide additional informations.
+    """
+
+    content = models.CharField(max_length=24)
```
Comment (marwaw) [inline game/models.py:74]: content or answer?

File: game/models.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    confirmation = models.TextField()
+    default = models.TextField()
+    rejections = HStoreField()
+
+    def evaluate(self, answer):
+        """
+        Evaluate if answer provided by user is correct and return appropriate answer
```
Comment (marwaw) [inline game/models.py]: proper answer sounds better

File: game/tests/conftest.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import pytest
+
+from game.models import Answer, Reward, Edition
+
+
+@pytest.fixture
+def sample_reward(db):
+    return Reward.objects.create(description='The cake is a lie.')
```
Comment (marwaw) [inline game/tests/conftest.py]: get_or_create

File: game/tests/conftest.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+
+@pytest.fixture
+def sample_answers(db):
+    return [Answer.objects.create(content='a',
+                                  confirmation='Congratulations! Your answer is correct!',
+                                  default="Sorry, it's not correct",
+                                  rejections={'aa': 'Overcomplicated',
+                                              'z': 'Nope.'})
+            for _ in range(2)]
```
Comment (marwaw) [inline game/tests/conftest.py:23]: Two the same sample answers

**Review by marwaw — CHANGES_REQUESTED** [review]
- think about develop branch
- make better documentation

File: game/models.py (hunk 1 above)
Comment (gchilczuk) [reply game/models.py:16]: I don't have a better idea how to name it :/

File: game/models.py (hunk 5 above)
Comment (gchilczuk) [reply game/models.py:74]: I don't like when the object `Answer` has an attribute called `answer`.  I don't have any better name

**Pull request merged**